### PR TITLE
Enable policy_document from pillars

### PR DESCRIPTION
**Pull request by vivianho**
### What does this PR do?
- Changed some log.debugs to log.errors
- Add another parameter to boto_iam_role.present to retrieve policy documents from pillars.
### What issues does this PR fix or reference?

N/A
### New Behavior

Users can manage an iam role policy document using pillars.
### Tests written?

No

### CODE REVIEW

File: salt/states/boto_iam_role.py [hunk 1]
```diff
@@ -123,6 +124,11 @@ def present(
     policy_document
         The policy that grants an entity permission to assume the role. (See https://boto.readthedocs.io/en/latest/ref/iam.html#boto.iam.connection.IAMConnection.create_role)
 
+    policy_document_from_pillars
+        A pillar key that contains a role policy document. The statements
+        defined here will be appended with the policy document statements
+        defined in the policy_document argument.
+
```
Comment (rallytime) [inline salt/states/boto_iam_role.py]: Can you add a `.. versionadded:: Nitrogen` tag here for this new option?

File: salt/modules/boto_iam.py [hunk 2]
```diff
@@ -1737,7 +1738,7 @@ def delete_policy(policy_name,
         conn.delete_policy(policy_arn)
         log.info('Deleted {0} policy.'.format(policy_name))
     except boto.exception.BotoServerError as e:
-        aws = salt.utils.boto.get_error(e)
+        aws = __utils__['boto.get_error'](e)
```
Comment (rallytime) [inline salt/modules/boto_iam.py]: Good catch on changing these!

**Review by rallytime — COMMENTED** [review]
Thanks for submitting this! I had one very minor comment, otherwise this looks good to me.

@ryan-lane or @kraney Do either of you have anything to add?

**Comment by ryan-lane** [comment]
Nothing to add. We already reviewed this internally at lyft :)

**Comment by rallytime** [comment]
Oh, great! Thanks everyone!

**Comment by rallytime** [comment]
I can pull in the versionadded change. Let's get this in. Thanks again @vivianho!

**Comment by kraney** [comment]
No comments from me. Looks like a nice add.

**Pull request merged**